### ISSUE TITLE
Wallabag container fails as quadlet but successful with podman run

### ISSUE THREAD

**Issue by tfranken90** [containers/podman#17572]
### Issue Description

This podman command runs successfully

`podman run -p 8181:80 -e "SYMFONY__ENV__DOMAIN_NAME=http://localhost:8181" docker.io/wallabag/wallabag:latest`

However, this quadlet file doesn't work

[Container]
Image=docker.io/wallabag/wallabag:latest
ContainerName=wallabag
PublishPort=8181:80
Environment="SYMFONY__ENV__DOMAIN_NAME=http://localhost:8181"

[Service]
Restart=always

### Steps to reproduce the issue

Steps to reproduce the issue
1. Create a container file under ~/.config/containers/systemd/wallabag.container
2. systemctl --user daemon-reload
3. systemctl --user start wallabag
4. Attempt to connect to service at localhost:8181

### Describe the results you received

journalctl reports this error (although the service does not fail)

`Feb 20 02:33:16 fedora wallabag[166752]: nginx: [alert] could not open error log file: open() "/var/lib/nginx/logs/error.log" failed (6: No such device or address)`

### Describe the results you expected

I expected the same results as the podman command. 

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.29.0
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.5-1.fc37.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.5, commit: '
  cpuUtilization:
    idlePercent: 82.95
    systemPercent: 5.32
    userPercent: 11.72
  cpus: 4
  distribution:
    distribution: fedora
    variant: silverblue
    version: "37"
  eventLogger: journald
  hostname: fedora
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.1.11-200.fc37.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 212901888
  memTotal: 8195121152
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.8-1.fc37.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.8
      commit: 0356bf4aff9a133d655dc13b1d9ac9424706cac4
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-8.fc37.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 5952237568
  swapTotal: 8194617344
  uptime: 45h 28m 24.00s (Approximately 1.88 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /var/home/tom/.config/containers/storage.conf
  containerStore:
    number: 12
    paused: 0
    running: 6
    stopped: 6
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/tom/.local/share/containers/storage
  graphRootAllocated: 248338448384
  graphRootUsed: 200945606656
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 60
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /var/home/tom/.local/share/containers/storage/volumes
version:
  APIVersion: 4.4.1
  Built: 1675940333
  BuiltTime: Thu Feb  9 04:58:53 2023
  GitCommit: ""
  GoVersion: go1.19.5
  Os: linux
  OsArch: linux/amd64
  Version: 4.4.1
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

I'm running Fedora Silverblue 37

**Comment by rhatdan** [comment]
Could you use quadlet --dryrun on the unit file to see if it shows you an issue?

**Comment by rhatdan** [comment]
/usr/libexec/podman/quadlet --dryrun

**Comment by tfranken90** [comment]
I get this error when using dryrun

`quadlet-generator[205790]: No files to parse from [/etc/containers/systemd /usr/share/containers/systemd]`

How do I specify the unit/where is it stored by quadlet?

**Comment by rhatdan** [comment]
You have to move the quadlet file to one of those directories, if you are running in rootful mode.  If you are running in rootless mode then the file should be moved to $HOME/.config/containers/systemd

**Comment by vrothberg** [comment]
Run `/usr/libexec/podman/quadlet -dryrun -user` .   The `-user` flag will look into HOME

**Comment by tfranken90** [comment]
Hm, I'm running rootless and the wallabag.container file is in that directory, but no luck. I went ahead and copied it into the /etc location and ran with sudo. Here's the output. 


---wallabag.service---
[X-Container]
Image=docker.io/wallabag/wallabag:latest
ContainerName=wallabag
PublishPort=8181:80
Environment="SYMFONY__ENV__DOMAIN_NAME=http://localhost:8181"

[Service]
Restart=always
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStopPost=-/usr/bin/podman rm -f -i --cidfile=%t/%N.cid
ExecStopPost=-rm -f %t/%N.cid
Delegate=yes
Type=notify
NotifyAccess=all
SyslogIdentifier=%N
ExecStart=/usr/bin/podman run --name=wallabag --cidfile=%t/%N.cid --replace --rm --log-driver passthrough --runtime /usr/bin/crun --cgroups=split --sdnotify=conmon -d --publish 8181:80 --env SYMFONY__ENV__DOMAIN_NAME=http://localhost:8181 docker.io/wallabag/wallabag:latest

[Unit]
SourcePath=/etc/containers/systemd/wallabag.container
RequiresMountsFor=%t/containers

**Comment by rhatdan** [comment]
The reason quadlet is failing is it does not see the unit file. In the example above, if you start the service, does it run?


Here is what I see in the homedir.

Here is what I see when I run in rootless mode.

$ /usr/libexec/podman/quadlet --user --dryrun
quadlet-generator[280323]: Loading source unit file /home/dwalsh/.config/containers/systemd/mysleep.container
...

**Comment by Luap99** [comment]
I think this could be caused by `--log-driver passthrough`, maybe try changing it to journald.
Also could you check for selinux errors? `sudo ausearch -m avc -ts recent`
I know the nginx images typically link `/var/lib/nginx/logs/error.log` to `/dev/stderr` so it tries to open stderr again and this fails for some reason.

**Comment by tfranken90** [comment]
@Luap99 I think you are write about nginx has journalctl prints a continuous `nginx: [alert] could not open error log file: open() "/var/lib/nginx/logs/error.log" failed (6: No such device or address)`

`sudo ausearch -m avc -ts recent` returned `<no matches>` 

How do I change the log-driver in the quadlet file?

**Comment by tfranken90** [comment]
Adding this line `PodmanArgs=--log-driver journald` to my quadlet file fixed the problem. thanks!

**Comment by vrothberg** [comment]
@rhatdan @alexlarsson @ygalblum PTAL

Should we change the default log-driver of Quadlet?  Currently, it's setting `passthrough` unless the user explicitly sets a one in the `.container` file. I have growing feeling that Quadlet should probably just use the system defaults which can be set in `containers.conf`.  Alternatively, a `log-driver-quadlet` option could be added to `containers.conf`.  Some global setting somewhere.

**Comment by ygalblum** [comment]
There was a reason why we set the default log-driver to `passthrough`. The original code had this comment:
```
// When running under Systemd use passthrough as the default log-driver.
// When doing so, the journal socket is passed to the containers as-is which has two advantages:
// 1. journald can see who the actual sender of the log event is,
//    rather than thinking everything comes from the conmon process
// 2. conmon will not have to copy all the log data
```
So, the reason for using `passthrough` was not because of Quadlet, but rather because of running under `systemd`. The reason we reverted this change and made it specific for Quadlet was backward compatibility.

Also note that for 4.5, we already have a dedicated key for the LogDriver (for both `.container` and `.kube`) along with the documentation stating that the default is in fact `passthrough` (which I think could have helped in this case).

Furthermore, I think there are different considerations when running under `systemd`. So, while we can have Quadlet use the default from `containers.conf`, I don't see anyone changing it to `passthrough` as it will require anyone running a container from the command line to manually set the log-driver flag.

Having said that, Quadlet started as an opinionated solution and I think we've removed most of its opinions, so what's another one.

We can have it not set anything unless the key is explicitly set
